mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -558,22 +558,36 @@ class QuantizeLinearConverter final
     Type elementType = getShapedElementTy(input);
     Value inverseScale =
         createOpAndInfer<tosa::ReciprocalOp>(rewriter, loc, elementType, scale);
-    if (getShapedElementTy(inverseScale) != getShapedElementTy(input)) {
-      inverseScale = createCastOp(rewriter, loc, elementType, inverseScale);
-    }
```

Comment:
> Remove handling for input/output type != scale type, since that can't happen in MIGraphX IR

I could be wrong, but I remember (either by looking at migraphx code base, or referenced by @pfultz2) that it is entirely possible to have fp32 to be the type of scale, which becomes different than input in quantizelinear or output in dequantizelinear.

Reply:
@pfultz2 specifically told me that the input and the scale have to have the same type in MIGraphX IR yesterday

Reply:
Yes the input and scale should be the same type.